libcontainer/cgroups/systemd/v1.go

```diff
@@ -353,15 +353,18 @@ func (m *legacyManager) freezeBeforeSet(unitName string, r *configs.Resources) (
 		// a non-existent unit returns default properties,
 		// and settings in (2) are the defaults.
 		//
-		// Do not return errors from getUnitProperty, as they alone
+		// Do not return errors from getUnitTypeProperty, as they alone
 		// should not prevent Set from working.
-		devPolicy, e := getUnitProperty(m.dbus, unitName, "DevicePolicy")
-		if e == nil && devPolicy.Value == dbus.MakeVariant("auto") {
-			devAllow, e := getUnitProperty(m.dbus, unitName, "DeviceAllow")
-			if e == nil && devAllow.Value == dbus.MakeVariant([]deviceAllowEntry{}) {
-				needsFreeze = false
-				needsThaw = false
-				return
+		unitType := getUnitType(unitName)
+		devPolicy, e := getUnitTypeProperty(m.dbus, unitName, unitType, "DevicePolicy")
+		if e == nil && devPolicy.Value.String() == `"auto"` {
```

Comment:
interesting ... comparing with `"auto"` instead of `auto`

Reply:
This is because `(dbus.Variant).String` uses `strconv.Quote` to format strings.

I am not quite sure whether the old or the new way to compare is better... Old one is working fine (for `devPolicy`, that is), so maybe leave it as is.